xpti/include/xpti/xpti_data_types.h

```diff
@@ -386,20 +386,46 @@ enum class trace_point_type_t : uint16_t {
   /// Used to notify that memory has been released.
   mem_release_end = XPTI_TRACE_POINT_END(19),
   /// Used to notify that offload buffer will be created
-  offload_alloc_construct = XPTI_TRACE_POINT_BEGIN(20),
+  offload_alloc_buffer_construct = XPTI_TRACE_POINT_BEGIN(20),
   /// Used to notify about association between user and internal
   /// handle of the offload buffer
-  offload_alloc_associate = XPTI_TRACE_POINT_BEGIN(21),
+  offload_alloc_buffer_associate = XPTI_TRACE_POINT_BEGIN(21),
   /// Used to notify that offload buffer will be destructed
-  offload_alloc_destruct = XPTI_TRACE_POINT_BEGIN(22),
+  offload_alloc_buffer_destruct = XPTI_TRACE_POINT_BEGIN(22),
```

Comment:
Since we are in an ABI breakage window, can you change this to:
`offload_alloc_buffer_destruct = XPTI_TRACE_POINT_END(20),`

All pairs, such as [*_begin, *_end], [*_construct, *_destruct] etc should use XPTI_TRACE_POINT_BEGIN(X)/XPTI_TRACE_POINT_END(X) pairs where the same ID is used for the pair  - in this case X.

We have 64 trace _BEGIN/_END pairs we can define here as default. We can extend it with user defined trace types (see [documentation](https://github.com/intel/llvm/blob/sycl/xptifw/doc/XPTI_Framework.md#xptiregisteruserdefinedtracepoint)). So, we need to be careful about assigning trace_type IDs. If they are truly not scoped pair of trace types, then we may run out of IDs and may have to extend it to use more bits.

So, please consider:
`offload_alloc_buffer_associate = XPTI_TRACE_POINT_BEGIN(21),`
`offload_alloc_buffer_release = XPTI_TRACE_POINT_END(21),`